src/borg/key.py

```diff
@@ -57,6 +59,75 @@ class RepoKeyNotFoundError(Error):
     """No key entry found in the config of repository {}."""
 
 
+class SignatureError(Error):
+    """Invalid signature for {}."""
+
+
+class SignedFile(FileLikeWrapper):
+    # generic enough that it can be used without a KeyBase
```

Comment:
put it into helpers.py? I'ld expect some trouble with circular dependencies when put here.